tests/unit_tests/tools/test_linker.py

```diff
@@ -350,3 +369,51 @@ def test_linker_inheriting():
     with pytest.raises(RuntimeError) as err:
         linker_mpif90.get_lib_flags("does_not_exist")
     assert "Unknown library name: 'does_not_exist'" in str(err.value)
+
+
+def test_linker_profile_flags_inheriting(mock_c_compiler):
+    '''Test nested compiler and nested linker with inherited profiling flags.
+
+    '''
+    mock_c_compiler_wrapper = CompilerWrapper(name="mock_c_compiler_wrapper",
+                                              compiler=mock_c_compiler,
+                                              exec_name="exec_name")
+    linker = Linker(mock_c_compiler_wrapper)
+    linker_wrapper = Linker(mock_c_compiler_wrapper, linker=linker)
+    count = 0
+    for compiler in [mock_c_compiler, mock_c_compiler_wrapper]:
+        compiler.define_profile("base")
+        compiler.define_profile("derived", "base")
+        compiler.add_flags(f"-f{count}", "base")
+        compiler.add_flags(f"-f{count+1}", "derived")
+        count += 2
+
+    # One set f1-f4 from the compiler wrapper, one from the wrapped linker
```

Comment:
I think this comment should be the following, based on list in the assertion?

```suggestion
    # One set f0-f3 from the compiler wrapper, one from the wrapped linker
```